pandas/io/excel/_base.py

```diff
@@ -809,14 +820,28 @@ class ExcelFile:
         "pyxlsb": _PyxlsbReader,
     }
 
+    # Excel templates are not supported by read_excel according to the
+    #  documentation, but openpyxl specifies them as supported.
+    _supported_engine_filetypes = {
+        "xlrd": ["xls", "xlsx", "xlsm"],
+        "openpyxl": ["xlsx", "xlsm", "xltx", "xltm"],
+        "odf": ["odf", "ods", "odt"],
+        "pyxlsb": ["xlsb"],
+    }
+
```

Comment:
I'd like to avoid this if possible. I don't think we should be saying what file types an engine should be supporting. Rather, we should try to do what the user requests, and propagate the engine's error message if that's not possible.

Reply:
I agree. It was a longshot. I'll revert these changes and document them accordingly. My intention was to provide more feedback on why the engine did not work with the provided file, as the engines' messages are sometimes cryptic. Maybe it is a problem that is best solved on the engines themselves. I'll get on with it.